linuxkm/module_hooks.c

```diff
@@ -198,6 +198,58 @@ WC_MAYBE_UNUSED static int linuxkm_lkcapi_sysfs_deinstall_node(struct kobj_attri
     #include "linuxkm/lkcapi_glue.c"
 #endif
 
+int wc_linuxkm_check_for_intr_signals(void) {
+    static const int intr_signals[] = WC_LINUXKM_INTR_SIGNALS;
+    if (preempt_count() != 0)
+        return 0;
+
+#if defined(HAVE_FIPS) && defined(LINUXKM_LKCAPI_REGISTER)
+    /* ignore signals during FIPS startup sequence -- failed alg tests cause
+     * kernel panics on FIPS kernels.
+     */
+    if (linuxkm_lkcapi_registering_now)
+        return 0;
+#endif
+    if (signal_pending(current)) {
+        int i;
+        for (i = 0;
+             i < (int)sizeof(intr_signals) / (int)sizeof(intr_signals[0]);
+             ++i)
+        {
+            if (sigismember(&current->pending.signal, intr_signals[i])) {
+#ifdef WOLFSSL_LINUXKM_VERBOSE_DEBUG
+                pr_err("INFO: wc_linuxkm_check_for_intr_signals returning "
+                       "-EINTR on signal %d\n", intr_signals[i]);
+#endif
+                return INTERRUPTED_E;
+            }
+        }
+    }
+    return 0;
+}
+
+void wc_linuxkm_relax_long_loop(void) {
+#if WC_LINUXKM_MAX_NS_WITHOUT_YIELD >= 0
+    if (preempt_count() == 0) {
+#if (WC_LINUXKM_MAX_NS_WITHOUT_YIELD == 0) || !defined(CONFIG_SCHED_INFO)
```

Comment:
nit: indenting this nested #if else endif clause would be more readable.

Reply:
no doubt.  fixed.